backend/inspirehep/snow/api.py

```diff
@@ -0,0 +1,645 @@
+
+    def get_functional_categories(self, query=None, fields=None):
+        """Returns list of all functional categories.
+
+        Args:
+            query (str): query string for filtering functional categories.
+            fields (str): list of fields to be returned in a search result.
+
+        Returns:
+            :obj:`list` of :obj:`dict`: list of all functional categories with requested fields.
+        """
+
+        return self.search("u_functional_element_category", query, fields)
+
+    def get_functional_categories_from_cache(self, force_update=False):
+        """Returns list of all functional categories.
+
+        Args:
+            force_update (bool): should cache entry be updated.
+
+        Returns:
+            :obj:`dict`: Dictionary of functional categories as {category_id: category_data}.
+        """
+        categories = current_cache.get("snow_functional_categories")
+        if not categories or force_update:
+            new_categories = {
+                category["sys_id"]: category
+                for category in self.get_functional_categories()
+            }
+            current_cache.set(
+                "snow_functional_categories",
+                new_categories,
+                timeout=current_app.config.get("SNOW_CACHE_TIMEOUT", 86400),
+            )
+            return new_categories
+        return categories
+
+    def get_users(self, query=None, fields=None):
+        """Get SNOW users.
+
+        Args:
+            query (str): query to filter the search result.
+            fields (str): comma-separated list of relevant fields.
+
+        Returns:
+            :obj:`list` of :obj:`dict`: snow users matching the query string.
+        """
+        return self.search("sys_user_grmember", query, fields)
+
+    def get_users_from_cache(self, force_update=False):
+        """Returns list of all users from cache.
+        Args:
+            force_update (bool): should cache entry be updated
+
+        Returns:
+            dict of users as {user_id: user_data}
+        """
+        users = current_cache.get("snow_users")
+        if not users or force_update:
+            new_users = {user["user.sys_id"]: user for user in self.get_users()}
+            current_cache.set(
+                "snow_users",
+                new_users,
+                timeout=current_app.config.get("SNOW_CACHE_TIMEOUT", 86400),
+            )
+            return new_users
+        return users
+
+
+class InspireSnow(SnowTicketAPI):
+    """Implements Inspire-specific methods to manage tickets related to Inspire."""
+
+    def __init__(
+        self,
+        url=None,
+        client_id=None,
+        secret=None,
+        auth_url=None,
+        verify_cert=False,
+        inspire_fe_id=None,
+        inspire_user_id=None,
+    ):
+        super().__init__(url, client_id, secret, auth_url, verify_cert)
+        self.inspire_fe_id = inspire_fe_id or current_app.config.get("SNOW_INSPIRE_FE")
```

Comment:
small thing, because we are going to forget what `fe` means, can we use the full name `functional_element`? Also we can drop `inspire_` prefix from vars, it's already clear from the classname that we are using it for INSPIRE project